version: "3.9"
   
services:
  db:
    image:  postgres:14-alpine
    volumes:
      - ./data/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=library_db
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
    ports:
      - 5432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      timeout: 45s
      interval: 10s
      retries: 10
    restart: always
    container_name: db_container
    networks:
      back_net:
        ipv4_address: 192.168.0.2

  web:
    build:
      context: library_back
    command: bash  -c "
        python manage.py makemigrations
        && python manage.py migrate
        && python manage.py runserver 0.0.0.0:8000
      "
    volumes:
      - ./library_back:/app
    environment:
      - POSTGRES_HOST=db
    ports:
      - "8000:8000"
    depends_on:
      - db
    container_name: web_container
    networks:
      back_net:
        ipv4_address: 192.168.0.3

networks:
  back_net:
    ipam:
      driver: default
      config:
      - subnet: "192.168.0.0/24"
        gateway: 192.168.0.1

    


